python/simulation.py

```diff
@@ -2234,6 +2247,20 @@ def add_source(self, src):
                 add_eig_src()
             else:
                 add_eig_src(src.amp_func)
+        elif isinstance (src, GaussianBeamSource):
+            gaussianbeam_args = [
+                py_v3_to_vec(self.dimensions, src.beam_x0, is_cylindrical=self.is_cylindrical),
+                py_v3_to_vec(self.dimensions, src.beam_kdir, is_cylindrical=self.is_cylindrical),
+                src.beam_w0,
+                src.src.swigobj.frequency().real,
+                self.fields.get_eps(py_v3_to_vec(self.dimensions, src.center, self.is_cylindrical)).real,
+                self.fields.get_mu(py_v3_to_vec(self.dimensions, src.center, self.is_cylindrical)).real,
+                [src.beam_E0.x, src.beam_E0.y, src.beam_E0.z]
```

Comment:
This is a list, but your typecheck is expecting a numpy array? It might still work but you should either convert to numpy array or do a `PyList_Check()` instead of `is_array()`...

Reply:
The typemaps know how to convert lists IIRC.